cli/rack/__init__.py

```diff
@@ -350,12 +350,169 @@ def utility_copygraph_driver(base_url: Url, triple_store: Optional[Url], triple_
     semtk3.set_host(base_url)
     triple_store = triple_store or DEFAULT_TRIPLE_STORE
     triple_store_type = triple_store_type or DEFAULT_TRIPLE_STORE_TYPE
-    
+
     @with_status(f'Copying {str_highlight(from_graph)} to {str_highlight(to_graph)}')
     def go() -> dict:
         return semtk3.copy_graph(from_graph, to_graph, triple_store, triple_store_type, triple_store, triple_store_type)
     go()
 
+class IngestionBuilder:
+    def __init__(self) -> None:
+        self.fresh: int = 0
+        self.model_graphs: Set[str] = set()
+        self.data_graphs: Set[str] = set()
+        self.manifests: Set[Path] = set()
+
+    def next_fresh(self) -> int:
+        result = self.fresh
+        self.fresh = result + 1
+        return result
+
+    def model(
+        self,
+        from_path: Path,
+        to_path: Path,
+    ) -> None:
+        with open(from_path, mode='r', encoding='utf-8-sig') as f:
+            obj = yaml.safe_load(f)
+
+        frombase = from_path.parent
+        tobase = to_path.parent
+
+        files = obj['files']
+        for (i,file) in enumerate(files):
+            file = Path(file)
+            shutil.copyfile(frombase.joinpath(file), tobase.joinpath(file.name), follow_symlinks=True)
+            files[i] = file.name
+
+        c = obj.get('model-graphs')
+        if c is None:
+            self.model_graphs.add(str(MODEL_GRAPH))
+        elif type(c) == str:
```

Comment:
Python code usually uses `isinstance` rather than a direct type comparison in order to accomodate subclasses.  For example, if yaml decided to subclass str at some point to add metadata about source location or something like that.